app/src/main/java/dev/arkbuilders/arkmemo/models/TextNote.kt

```diff
@@ -1,20 +1,13 @@
 package dev.arkbuilders.arkmemo.models
 
 import android.os.Parcelable
-import kotlinx.parcelize.IgnoredOnParcel
 import dev.arkbuilders.arkmemo.data.ResourceMeta
 import kotlinx.parcelize.Parcelize
 
 @Parcelize
 data class TextNote (
+    val title: String,
+    val description: String = "",
     val content: Content,
-    @IgnoredOnParcel
-    var meta: ResourceMeta? = null
-): Parcelable
-{
-    @Parcelize
-    data class Content(
-        val title: String,
-        val data: String
-        ): Parcelable
-}
\ No newline at end of file
+    private val meta: ResourceMeta? = null
+): BaseNote(title, description, content, meta), Parcelable
```

Comment:
Please avoid using base class type. Use single class only. Please conform to the principle: "Composition over inheritance"

Reply:
@ShubertMunthali he is very right not. Although we use `Base` classes sometimes to reduce amount of boilerplate, composition/aggregation is often better. We should review inheritance usage in `arklib-android`, too.

Reply:
By the way, we might need to have separate variants of a `Note` class for text and graphic notes. But I need to build mental model of current codebase to say more.

Reply:
Alright, we will use composition then

Reply:
> By the way, we might need to have separate variants of a `Note` class for text and graphic notes. But I need to build mental model of current codebase to say more.

What if we create a ```Note```  interface?